xarray/core/dataset.py

```diff
@@ -4173,14 +4173,6 @@ def update(self, other: "CoercibleMapping") -> "Dataset":
             - mapping {var name: (dimension name, array-like)}
             - mapping {var name: (tuple of dimension names, array-like)}
 
-        Returns
-        -------
-        updated : Dataset
-            Updated dataset. Note that since the update is in-place this is the input
-            dataset.
-
-            It is deprecated since version 0.17 and scheduled to be removed in 0.19.
```

Comment:
I wonder if this one in particular is worth the hassle. We don't have any way of issuing a warning here, so users will just get an error.

Instead, we might just remove the documentation but keep returning `self`.

Reply:
we discussed this in #4932. I'm not quite sure about keeping the return value. I see your point, but it's also pretty easy to forget that `.update` is a in-place operation if it returns something (which means that the use of the return value of `.update` might actually indicate a bug). Do we have a not-inplace version of `.update` we could recommend instead? I.e. does `.assign(ds)` do the same as `.update(ds)` except the original `Dataset` is not modified?

Reply:
I'm not sure what the best answer is, but would very much like to get the rest of this PR merged so that we can issue the release!

Reply:
`merge()` is basically the same as `update()` except it's not in-place.

Reply:
I'll point to that in the docstring.

What should we do with the return value? It's not too difficult to add it back, but I think we should drop it eventually.